account/account.go

```diff
@@ -13,5 +13,31 @@ type Account interface {
 	// GetPChainAddress returns the P-Chain address for the given network
 	GetPChainAddress(network network.Network) (string, error)
 
+	// GetXChainAddress returns the X-Chain address for the given network
+	GetXChainAddress(network network.Network) (string, error)
+
+	// GetCChainAddress returns the C-Chain address (same as EVM address)
+	GetCChainAddress() (string, error)
+
+	// GetEVMAddress returns the EVM address (0x format)
+	GetEVMAddress() (string, error)
+
 	GetKeychain() (*secp256k1fx.Keychain, error)
 }
+
+// AccountInfo contains public information about an account
+// Does NOT expose private key material
+type AccountInfo struct {
+	Name       string // Account name
+	PAddress   string // P-Chain address
+	XAddress   string // X-Chain address
+	CAddress   string // C-Chain address
+	EVMAddress string // EVM address (0x format)
+}
+
+// AccountSpec contains account creation/import specifications
+type AccountSpec struct {
+	PrivateKey     string // Hex-encoded private key (for local accounts)
+	DerivationPath string // For Ledger accounts
+	RemoteKeyID    string // For remote signing services
+}
```

Comment:
Same here, seems that each field is specific to each account type like local accountn, server wallet, etc. They can be localized to each account type. like private key can be private getter function in local account

Reply:
I dont think it is, adr 5 discussion was all around evm and p/x/c placements in wallet interface

Reply:
same